### ISSUE TITLE
[NFR] Form and Model Validation Error Rendering

### ISSUE THREAD

**Issue by jymboche** [phalcon/cphalcon#1891]
A common feature in user-friendly web forms is to display the field's error/validation messages directly below the field. Also, its good practice to add a css error class to highlight which fields require attention. Looking through the sample phalcon apps, it seems they all display errors only above the form. I believe this is because there isn't a good way to do it currently in the framework.

If using Phalcon\Forms\Form it already has this functionality to check for errors (Form::hasMessagesFor) and get them (Form::getMessagesFor). However, this only seems to work if validation is done within the form. If validation is done in the model, the form is unaware of the messages (and there is no way to inject them). Consequently, checking for validation errors and rendering them becomes cumbersome.

There are several potential options for making this easier. First is to add the methods hasMessagesFor and getMessagesFor to Phalcon\Mvc\Model. Which could be used like this:

``` php
// @var $user Model
$class = $user->hasMessagesFor('username')? " error" : "no-error";
<div class='field<?php echo $class; ?>'>
    <?php echo $tag->textField('username'); ?>
    <?php echo $user->getMessagesFor('username'); ?>
</div>
```

Or a separate class could be added to handle Model validation messages. Yii's CActiveForm here is an example:
http://www.yiiframework.com/doc/api/1.1/CActiveForm

Which works like this:

``` php
$form = $this->beginWidget('CActiveForm');
$form->errorSummary($model); // prints <ul> of all model errors
$form->textField($model, 'username'); // renders textField
$form->error($model, 'username'); // prints validation message as string
$form->labelEx($model, 'username');
// calls Model::attributeLabels() to get field labels and render them (or use default column name)
```

In conclusion, user-friendly forms should definitely highlight invalid fields and optionally render messages below them/beside them. Any thoughts?
##
